src/Renderer/ThreeExtended/Feature2Mesh.js

```diff
@@ -42,49 +140,77 @@ function coordinatesToMesh(coordinates) {
             break;
         }
         case 'polygon': {
-            const indices = [];
+            let offset = 0;
             /* eslint-disable guard-for-in */
             for (const id in coordinates.featureVertices) {
-                const polygon = coordinates.featureVertices[id];
-                const contour = vertices.slice(polygon.offset * 3, (polygon.offset + polygon.count) * 3);
-                const triangles = Earcut(contour, null, 3);
+                // extract contour coodinates and properties of one feature
+                const { contour, propertie } = extractFeature(coordinates, properties, id);
+                // get altitude and extrude amount from properties
+                const altitudeBottom = style.altitude ? style.altitude(propertie) : 0;
+                const extrudeAmount = style.extrude ? style.extrude(propertie) : 0;
```

Comment:
I'm not a huge fan of this design. Here's what I had in mind:
- add a callback in the layer, called for each new feature. Actually this callback exists - but is misnamed `colorFunction`
- expose some helper functions to perform feature polygon extrusion easily



This way, you won't have to complexify even more this part and the user is free to implement whatever feature.

Reply:
I don't think it's a good idea to do the extrusion in a callback because
- do the extrusion means creating a mesh
- at this point (in the callback), the mesh is already created. So I will have to destroy the mesh and create another one ?? It's not a good idea.
- and Feature2Mesh seems to be the good place to convert a feature to a mesh.

If you think it's complex, I can extract methods.

Reply:
I don't think it's a good idea to do the extrusion in a callback because
- do the extrusion means creating a mesh
- at this point (in the callback), the mesh is already created. So I will have to destroy the mesh and create another one ?? It's not a good idea.
- and Feature2Mesh seems to be the good place to convert a feature to a mesh.

If you think it's complex, I can extract methods.

Reply:
I don't think it's a good idea to do the extrusion in a callback because
- do the extrusion means creating a mesh.
- at this point (in the callback), the mesh is already created. So I will have to destroy the mesh and create another one ?
- and Feature2Mesh seems to be the good place to convert a feature to a mesh.

If you think it's complex, I can extract methods.

Reply:
The problem is you're burying deep inside iTowns something the user might want to control precisely.
IMHO the callback should be right here:
https://github.com/iTowns/itowns/pull/519/files#diff-b1820acc0c4f07c7e51a605b08ae2cc1R101

This code could become something like:
```js
callback(GeoJSON2Feature.parse(crs, geojson, tile.extent));
```

And then the user could do whatever fits its usecase:
- convert to 2D mesh (using e.g `Feature2Mesh.convert`)
- convert to extruded mesh (using a new function: `Feature2Mesh.convertAndExtrude` or whatever)
- something entirely different (build its own meshes, filter heavily, etc etc)

Reply:
ah ok, I understand. I try this approach.

Reply:
I've updated the code to extract a callback to choose extrusion of not.
But I still need to get altitude and and extrude amount from properties.